HW7(Tests)/Functions.test.js

```diff
@@ -0,0 +1,167 @@
+const {describe, test, expect, beforeAll} = require('@jest/globals');
+const {validateEmail, User, getUsers, getUsersEmails} = require('./Functions.js');
+
+describe('Testing function validateEmail with different emails.', () => {
+  describe('Should return true with valid emails.', () => {
+    const validEmailsToCheck = [
+      'example@gmail.com',
+      'EXAMPLE@GMAIL.COM',
+      'ExAmPlE@mail.ru',
+      '        example@gmail.com         ',
+      '        example@gmail.com',
+      'example@gmail.com         ',
+      'exmpl@mail.ru',
+      'example12@gmail.com',
+      '12example@gmail.com',
+    ];
+
+    test('Returns true for valid emails.', () => {
+      for (const email of validEmailsToCheck) {
+        expect(validateEmail(email)).toBe(true);
+      }
+    });
+  });
+
+  describe('Should return false with invalid emails.', () => {
+    const invalidEmailsToCheck = [
+      'exmp@gmail.ru',
+      'exm@mail.ru',
+      'example @gmail.com',
+      'example@gmail. com',
+      'example@mailru',
+      'exmple@mailru.',
+      '.example@mailru',
+      'exm@.mailru',
+      'exm@.',
+      'examplemail.ua',
+      '@examplemail.ua',
+      'examplemail.ua@',
+      'example@12gmail.com',
+      'example@gmail.12',
+      'ex!ample@gmail.com',
+      'ex?ample@gmail.com',
+      'ex,ample@gmail.com',
+      'ex@..a@mple@gmail.com',
+      'ex.ample@gmail.com'
+    ];
+
+    test('Returns false for invalid emails.', () => {
+      for (const email of invalidEmailsToCheck) {
+        expect(validateEmail(email)).toBe(false);
+      }
+    });
+  });
+});
+
+describe('Testing class User.', () => {
+  describe('Passing both arguments to the User constructor.', () => {
+    const user = new User('Dmytro', 'Khyzhniak');
+
+    test('Returned object properties and values are correct.', () => {
+      expect(user).toEqual({
+        firstName: 'Dmytro',
+        lastName: 'Khyzhniak'
+      });
+    });
+
+    test('Method getFullName() returns correct result.', () => {
+      expect(user.getFullName()).toBe('Dmytro Khyzhniak');
+    });
+  });
+
+  describe('Passing only one argument to the User constructor.', () => {
+    const user = new User('Dmytro');
+
+    test('Returned object properties and values are correct.', () => {
+      expect(user).toEqual({
+        firstName: 'Dmytro',
+        lastName: 'Simpson'
+      });
+    });
+
+    test('Method getFullName() returns correct result.', () => {
+      expect(user.getFullName()).toBe('Dmytro Simpson');
+    });
+  });
+
+  describe('Passing no arguments to the User constructor.', () => {
+    const user = new User();
+
+    test('Returned object properties and values are correct.', () => {
+      expect(user).toEqual({
+        firstName: 'Homer',
+        lastName: 'Simpson'
+      });
+    });
+
+    test('Method getFullName() returns correct result.', () => {
+      expect(user.getFullName()).toBe('Homer Simpson');
+    });
+  });
```

Comment:
in the test description it is better to add that we check the default values, so it will be clearer when you read the test without opening the source file